plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -823,6 +860,88 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         transaction = null;
     }
 
+    @Override
+    public void executeTableExecute(ConnectorSession session, ConnectorTableExecuteHandle tableExecuteHandle)
+    {
+        IcebergTableExecuteHandle executeHandle = (IcebergTableExecuteHandle) tableExecuteHandle;
+        switch (executeHandle.getProcedureId()) {
+            case EXPIRE_SNAPSHOTS:
+                executeExpireSnapshots(session, executeHandle);
+                return;
+            default:
+                throw new IllegalArgumentException("Unknown procedure '" + executeHandle.getProcedureId() + "'");
+        }
+    }
+
+    private void executeExpireSnapshots(ConnectorSession session, IcebergTableExecuteHandle executeHandle)
+    {
+        IcebergExpireSnapshotsHandle expireSnapshotsHandle = (IcebergExpireSnapshotsHandle) executeHandle.getProcedureHandle();
+
+        Table table = catalog.loadTable(session, executeHandle.getSchemaTableName());
+        int tableVersion = ((BaseTable) table).operations().current().formatVersion();
+        if (tableVersion > EXPIRE_SNAPSHOTS_MAX_SUPPORTED_TABLE_VERSION) {
```

Comment:
I'd call this `tableFormatVersion` version usually refers to the version number of the metadata being read.

Similarly use `format version` in the error message